Main/Source/miscitem.cpp

```diff
@@ -3806,3 +3810,147 @@ col16 trinket::GetMaterialColorC(int) const
   if(GetConfig() == POTTED_PLANT) { return MakeRGB16(0, 160, 0); }
   else { return MakeRGB16(0, 0, 0); }
 }
+
+void gastrap::StepOnEffect(character* Stepper)
+{
+  if(IsActive() && !IsBroken())
+  {
+    if(Stepper->IsPlayer())
+      ADD_MESSAGE("You step on %s.", GetExtendedDescription().CStr());
+    else if(Stepper->CanBeSeenByPlayer())
+      ADD_MESSAGE("%s steps on %s.", Stepper->CHAR_NAME(DEFINITE), GetExtendedDescription().CStr());
+
+    if(Stepper->IsPlayer())
+      game::AskForKeyPress(CONST_S("Trap activated! [press any key to continue]"));
+
+    if(GetSecondaryMaterial())
+    {
+      if(CanBeSeenByPlayer())
+        ADD_MESSAGE("It releases some gas!");
+
+      material* GasMaterial = GetSecondaryMaterial();
+      GetLevel()->GasExplosion(static_cast<gas*>(GasMaterial), GetLSquareUnder(), Stepper);
+    }
+    else
+    {
+      if(CanBeSeenByPlayer())
+        ADD_MESSAGE("Luckily, it's empty.");
+    }
+
+    if(!(RAND() % 10))
+    {
+      if(CanBeSeenByPlayer())
+        ADD_MESSAGE("%s gets jammed.", CHAR_NAME(DEFINITE));
+
+      SetIsActive(false);
+      Break(Stepper);
+    }
+  }
+}
+
+truth gastrap::ReceiveDamage(character* Damager, int Damage, int Type, int)
+{
+  if(Type & PHYSICAL_DAMAGE && Damage)
+  {
+    if(Damage > 125 || !(RAND() % (250 / Damage)))
+    {
+      if(GetSquareUnder()->CanBeSeenByPlayer(true))
+        ADD_MESSAGE("%s shatters!", GetExtendedDescription().CStr());
+
+      if(GetSecondaryMaterial())
+      {
+        if(CanBeSeenByPlayer())
+          ADD_MESSAGE("It releases some gas!");
+
+        material* GasMaterial = GetSecondaryMaterial();
+        GetLevel()->GasExplosion(static_cast<gas*>(GasMaterial), GetLSquareUnder(), Damager);
+      }
+      else
+      {
+        if(CanBeSeenByPlayer())
+          ADD_MESSAGE("Luckily, it's empty.");
+      }
+
+      RemoveFromSlot();
+      SendToHell();
+      return true;
+    }
+    else if(IsActive())
+    {
+      if(CanBeSeenByPlayer())
+        ADD_MESSAGE("%s gets jammed.", CHAR_NAME(DEFINITE));
+
+      SetIsActive(false);
+      Break(Damager);
+      return true;
+    }
+  }
+  else if(Type & (ENERGY|SOUND) && Damage)
+  {
+    if(Damage > 50 || !(RAND() % (100 / Damage)))
+    {
+      if(GetSquareUnder()->CanBeSeenByPlayer(true))
+        ADD_MESSAGE("%s shatters!", GetExtendedDescription().CStr());
+
+      if(GetSecondaryMaterial())
+      {
+        if(CanBeSeenByPlayer())
+          ADD_MESSAGE("It releases some gas!");
+
+        material* GasMaterial = GetSecondaryMaterial();
+        GetLevel()->GasExplosion(static_cast<gas*>(GasMaterial), GetLSquareUnder(), Damager);
+      }
+      else
+      {
+        if(CanBeSeenByPlayer())
+          ADD_MESSAGE("Luckily, it's empty.");
+      }
+
+      RemoveFromSlot();
+      SendToHell();
+      return true;
+    }
+  }
+  else
```

Comment:
@red-kangaroo try removing this `else` statement to prevent the error in Travis and just leave `return false` where it is, outside the curly braces.

Reply:
Travis is weird.